ios/REANodesManager.mm

```diff
@@ -211,14 +204,24 @@ - (instancetype)initWithModule:(REAModule *)reanimatedModule uiManager:(RCTUIMan
     _viewRegistry = [_uiManager valueForKey:@"_viewRegistry"];
     _shouldFlushUpdateBuffer = false;
   }
+#endif
 
-  _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(onAnimationFrame:)];
-  _displayLink.preferredFramesPerSecond = 120; // will fallback to 60 fps for devices without Pro Motion display
-  [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
-  [_displayLink setPaused:true];
+  __weak __typeof__(self) weakSelf = self;
+
+  RCTExecuteOnMainQueue(^void() {
```

Comment:
I feel like a better approach here would be to update displayLink usages to access it using getter. then in the getter we'd optionally initialize. It is important that the display link is both created and accessed on the UI thread. So we could also add some assertions to verify that. Schedulinng initialization may have its own problems as we don't want to guarantee that the field wouldn't be used before the task gets executed